src/Ilios/CoreBundle/Entity/Repository/CourseRepository.php

```diff
@@ -301,7 +271,165 @@ public function isUserInstructingInCourse(UserInterface $user, CourseInterface $
+    JOIN ilm_session_facet ilm ON ilm.session_id = s.session_id
+    JOIN ilm_session_facet_x_learner ilmxl ON ilmxl.ilm_session_facet_id = ilm.ilm_session_facet_id
+    JOIN user u ON u.user_id = ilmxl.user_id
+    WHERE u.user_id = :user_id
+  UNION
+  SELECT c.* FROM course c
+    JOIN `session` s ON s.course_id = c.course_id
+    JOIN ilm_session_facet ilm ON ilm.session_id = s.session_id
+    JOIN ilm_session_facet_x_group ilmxg ON ilmxg.ilm_session_facet_id = ilm.ilm_session_facet_id
+    JOIN `group` g ON g.group_id = ilmxg.group_id
+    JOIN group_x_user gxu ON gxu.group_id = g.group_id
+    JOIN user u ON u.user_id = gxu.user_id
+    WHERE u.user_id = :user_id
+  UNION
+  SELECT c.* FROM course c
+    JOIN `session` s ON s.course_id = c.course_id
+    JOIN offering o ON o.session_id = s.session_id
+    JOIN offering_x_instructor oxi ON oxi.offering_id = o.offering_id
+    JOIN user u ON u.user_id = oxi.user_id
+    WHERE u.user_id = :user_id
+  UNION
+  SELECT c.* FROM course c
+    JOIN `session` s ON s.course_id = c.course_id
+    JOIN offering o ON o.session_id = s.session_id
+    JOIN offering_x_instructor_group oxig ON oxig.offering_id = o.offering_id
+    JOIN `group` g ON g.group_id = oxig.instructor_group_id
+    JOIN group_x_user gxu ON gxu.group_id = g.group_id
+    JOIN user u ON u.user_id = gxu.user_id
+    WHERE u.user_id = :user_id
+  UNION
+  SELECT c.* FROM course c
+    JOIN `session` s ON s.course_id = c.course_id
+    JOIN ilm_session_facet ilm ON ilm.session_id = s.session_id
+    JOIN ilm_session_facet_x_instructor ilmxi ON ilmxi.ilm_session_facet_id = ilm.ilm_session_facet_id
+    JOIN user u ON u.user_id = ilmxi.user_id
+    WHERE u.user_id = :user_id
+  UNION
+  SELECT c.* FROM course c
+    JOIN `session` s ON s.course_id = c.course_id
+    JOIN ilm_session_facet ilm ON ilm.session_id = s.session_id
+    JOIN ilm_session_facet_x_instructor_group ilmxig ON ilmxig.ilm_session_facet_id = ilm.ilm_session_facet_id
+    JOIN `group` g ON g.group_id = ilmxig.instructor_group_id
+    JOIN group_x_user gxu ON gxu.group_id = g.group_id
+    JOIN user u ON u.user_id = gxu.user_id
+    WHERE u.user_id = :user_id
+) AS my_courses
+EOL;
 
+        $params = [];
+        $i = 0;
+        $sqlFragments = [];
+        foreach ($criteria as $name => $value) {
+            $i++;
+            if (! $meta->hasField($name)) {
+                throw new \Exception(sprintf('"%s" is not a property of the Course entity.', $name));
+            }
+
+            $column = $meta->getColumnName($name);
+            $label = 'param' . $i;
+            $params[$name] = $label;
+            if (is_array($value)) {
+                $sqlFragments[] = "{$column} IN (:{$label})";
+            } else {
+                $sqlFragments[] = "{$column} = :{$label}";
+            }
+        }
+        if (count($sqlFragments)) {
+            $sql .= ' WHERE ' . implode(' AND ', $sqlFragments);
+        }
+
+        if (is_array($orderBy)) {
```

Comment:
This isn't really necessary.  $orderBy has to be an array and if it is empty it is being forced  into an array.